src/main/java/org/folio/circulation/domain/RequestRules.java

```diff
@@ -0,0 +1,65 @@
+package org.folio.circulation.domain;
+
+import java.util.EnumMap;
+
+public class RequestRules {
+
+  private static EnumMap<ItemStatus, Boolean> recallRules;
+  private static EnumMap<ItemStatus, Boolean> holdRules;
+  private static EnumMap<ItemStatus, Boolean> pageRules;
+
+  static{
+    initRecallRules();
+    initHoldRules();
+    initPageRules();
+  }
+
+  private RequestRules(){
+    throw new IllegalStateException();
+  }
+
+  private static void initRecallRules(){
+    recallRules = new EnumMap<>(ItemStatus.class);
+    recallRules.put(ItemStatus.CHECKED_OUT, true);
+    recallRules.put(ItemStatus.AVAILABLE, false);
+    recallRules.put(ItemStatus.AWAITING_PICKUP, true);
+    recallRules.put(ItemStatus.IN_TRANSIT, true);
+    recallRules.put(ItemStatus.MISSING, false);
+    recallRules.put(ItemStatus.PAGED, false);
+    recallRules.put(ItemStatus.NONE, false);
+  }
+
+  private static void initHoldRules(){
+    holdRules = new EnumMap<>(ItemStatus.class);
+    holdRules.put(ItemStatus.CHECKED_OUT, true);
+    holdRules.put(ItemStatus.AVAILABLE, false);
+    holdRules.put(ItemStatus.AWAITING_PICKUP, true);
+    holdRules.put(ItemStatus.IN_TRANSIT, true);
+    holdRules.put(ItemStatus.MISSING, true);
+    holdRules.put(ItemStatus.PAGED, true);
+    holdRules.put(ItemStatus.NONE, true);
+  }
+
+  private static void initPageRules(){
+    pageRules = new EnumMap<>(ItemStatus.class);
+    pageRules.put(ItemStatus.CHECKED_OUT, false);
+    pageRules.put(ItemStatus.AVAILABLE, true);
+    pageRules.put(ItemStatus.AWAITING_PICKUP, false);
+    pageRules.put(ItemStatus.IN_TRANSIT, false);
+    pageRules.put(ItemStatus.MISSING, false);
+    pageRules.put(ItemStatus.PAGED, false);
+    pageRules.put(ItemStatus.NONE, false);
+  }
+
+  public static boolean canCreateRequestForItem(ItemStatus itemStatus, RequestType requestType){
+    if (requestType == RequestType.HOLD){
+      return holdRules.get(itemStatus);
+    } else if (requestType == RequestType.RECALL){
+      return recallRules.get(itemStatus);
+    } else if (requestType == RequestType.PAGE){
+      return pageRules.get(itemStatus);
+    } else {
+      return false;
+    }
+  }
```

Comment:
Maybe consider adding a 4th `EnumMap`, like `EnumMap<RequestType, EnumMap<ItemStatus, Boolean>> requestsRulesMap = new EnumMap<>(RequestType.class);` and store each rules map in this map. Then, instead of this if statement, you could do something like `return requestsRulesMap.get(requestType).get(itemStatus);`.

The `RequestType.NONE` type is a wrinkle, either we could have a `NONE` map with everything false or we could check for `null` on the get and return `false`. Seems like a `NONE` map would be better considering this should come from an API at some point.

Anyway, this is not a big deal and might be overkill. Okay, your got me... I just don't like if/else with enums (or in general)... though, for some reason, I don't mind a switch... :)

Reply:
I quite like that approach, as long as there is a fallback for non-existent keys for both maps.

The `none` request maybe could be `unknown` instead, or split into `unknown` and `invalid` (I haven't checked if the system distinguishes).

(I think switches can be cleaner, and can sometimes provide better hints that there is a missing abstraction. This is also where languages that have control structures like `when` can be more expressive).

Reply:
Adding 4th EnumMap as discussed here.